heroic-elasticsearch-utils/src/main/java/com/spotify/heroic/elasticsearch/SlowStartRateLimiter.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2015 Spotify AB.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.heroic.elasticsearch;
+
+import com.google.common.util.concurrent.RateLimiter;
+import com.spotify.heroic.time.Clock;
+import java.util.concurrent.TimeUnit;
+import lombok.Data;
+
+@Data
+public class SlowStartRateLimiter implements SimpleRateLimiter {
+    private final RateLimiter rateLimiter;
+    private final double ratePerSecond;
+    private final long slowStartDuration;
+    private final TimeUnit slowStartTimeUnit;
+    private final Clock clock;
+    private final long startTime;
+
+    /* Don't use an atomic type, to improve performance. The cost of this is that during startup
+     * there might be some extra slow path calls verifying the time in threads until all threads
+     * has seen the updated boolean value. No harm tho. This only affects startup performance and
+     * improves performance after. Repeated calls to setRate() are ok. */
+    private boolean isSlowStartMode;
```

Comment:
Right! Ha, true story, I actually thought of this needing to be volatile a couple of nights ago but forgot to write it down. Good catch.